var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -287,6 +287,11 @@ def test(self):
         args.append('-DAMReX_ROOT=' + self.prefix)
         args.append('-DMPI_C_COMPILER=' + self.spec['mpi'].mpicc)
         args.append('-DMPI_CXX_COMPILER=' + self.spec['mpi'].mpicxx)
+        if '+cuda' in self.spec:
+            args.append('-DCMAKE_CUDA_COMPILER='
+                        + self.spec['cuda'].prefix.bin
+                        + '/nvcc')
```

Comment:
Hm, I wonder if that works well on Cray where we often use `CC` (`CC -x cuda`) for CUDA compiler?

Reply:
for generic path joining, you can use `join_path`

Reply:
I did this because I was running into an issue where compiling the Cuda test with two different versions of the Cuda compiler was leading to errors. I am not sure if this is correct, but I my idea was to tell CMake to use the version of Cuda that originally compiled the installed version of AMReX. Is `join path` still the right thing in this case? 

About the Cray thing, how do you think I could test it? Would it be enough just test it on Perlmutter with the defaults, or would it be necessary to test a specific install configuration?

Reply:
I removed this section of code because its was only useful for getting the smoke test to run on Perlmutter. Therefore it was more about addressing some unique characteristics about the Perlmutter build environment rather than taking more general approach and letting users customize via other means.

Reply:
Yes, I think `join_path` would be better than `+ "/..."`